src/domTree.js

```diff
@@ -25,20 +25,73 @@ const createClass = function(classes) {
     return classes.join(" ");
 };
 
+class baseNode {
+    constructor() {
+        this.attributes = {};
+    }
+
+    attributesToNode(node) {
+        for (const attr in this.attributes) {
+            if (Object.prototype.hasOwnProperty.call(this.attributes, attr)) {
+                node.setAttribute(attr, this.attributes[attr]);
+            }
```

Comment:
This is really confusing.  Too many things are node here.  In this case the node being passed in a DOM node, right?

Reply:
I think ideally, if we using something like HAST to build a representation of the HTML we want using JS objects, we'd probably want to use the visitor pattern to build the actually DOM nodes.

Reply:
Not sure I got it about node. I've made it just to put common code here. Maybe the name `baseNode` is misleading?

But I got it about HAST. I guess you would want visitor now because I already see svgNode, pathNode and lineNode.

Reply:
Looking at the method `attributesToNode(node)` on the class `baseNode` I assumed that the `node` being passed in was an instance of `baseNode`.  Renaming the parameter to `domNode` would've helped.  Also, `attributesToNode` should probably be a helper function.

Reply:
Oh, I get it now. I'll move it to helper function and get rid of baseNode since this classes are going to change I guess.

Reply:
@spontaliku-softaria you may want to hold off on revisions to this PR.  I think we should do #800 first.

Reply:
OK, I'll stop spamming you with comments.

Reply:
The comments are fine.  I just don't think I'll be able to merge this right away and the changes proposed in #800 will require significant rework of this PR anyways.

Reply:
That's OK to me since I'm not fully in context of your plans. It's just we are experimenting with editor and I need to do all this anyway, here or in a private fork.